src/Microsoft.AspNetCore.Hosting/Internal/HostingApplication.cs

```diff
@@ -89,145 +49,22 @@ public Task ProcessRequestAsync(Context context)
-                exception: null,
-                formatter: HostingRequestStartingLog.Callback);
-        }
-
-        [MethodImpl(MethodImplOptions.NoInlining)]
-        private void LogRequestFinished(HttpContext httpContext, long startTimestamp, long currentTimestamp)
-        {
-            // IsEnabled isn't checked in the caller, startTimestamp > 0 is used as a fast proxy check
-            // but that may be because diagnostics are enabled, which also uses startTimestamp, so check here
-            if (_logger.IsEnabled(LogLevel.Information))
-            {
-                var elapsed = new TimeSpan((long)(TimestampToTicks * (currentTimestamp - startTimestamp)));
-
-                _logger.Log(
-                    logLevel: LogLevel.Information,
-                    eventId: LoggerEventIds.RequestFinished,
-                    state: new HostingRequestFinishedLog(httpContext, elapsed),
-                    exception: null,
-                    formatter: HostingRequestFinishedLog.Callback);
-            }
-        }
-
-        [MethodImpl(MethodImplOptions.NoInlining)]
-        private void RecordBeginRequestDiagnostics(HttpContext httpContext, long startTimestamp)
-        {
-            _diagnosticSource.Write(
-                DiagnosticsBeginRequestKey,
-                new
-                {
-                    httpContext = httpContext,
-                    timestamp = startTimestamp
-                });
-        }
-
-        [MethodImpl(MethodImplOptions.NoInlining)]
-        private void RecordEndRequestDiagnostics(HttpContext httpContext, long currentTimestamp)
-        {
-            _diagnosticSource.Write(
-                DiagnosticsEndRequestKey,
-                new
-                {
-                    httpContext = httpContext,
-                    timestamp = currentTimestamp
-                });
-        }
-
-        [MethodImpl(MethodImplOptions.NoInlining)]
-        private void RecordUnhandledExceptionDiagnostics(HttpContext httpContext, long currentTimestamp, Exception exception)
-        {
-            _diagnosticSource.Write(
-                DiagnosticsUnhandledExceptionKey,
-                new
-                {
-                    httpContext = httpContext,
-                    timestamp = currentTimestamp,
-                    exception = exception
-                });
-        }
-
-        [MethodImpl(MethodImplOptions.NoInlining)]
-        private static void RecordRequestStartEventLog(HttpContext httpContext)
-        {
-            HostingEventSource.Log.RequestStart(httpContext.Request.Method, httpContext.Request.Path);
+            _diagnostics.ContextDisposed(context);
         }
 
         public struct Context
         {
             public HttpContext HttpContext { get; set; }
             public IDisposable Scope { get; set; }
             public long StartTimestamp { get; set; }
+            public bool EventLogEnabled { get; set; }
+
+#if !NET451
```

Comment:
Meaningless? Aren't building for `NET451` only for `netstandard1.3` and `netstandard1.5`